--- webconfig/guinnux_web_base/srv/http/config/system_conf.lua	2015-12-15 10:48:10.692337698 +0200
+++ webconfig/guinnux_web_base/srv/http/config/system_conf-new.lua	2015-12-22 10:39:10.840265468 +0200
@@ -10,8 +10,8 @@
 
 top_proc = "top -b -n1"
 
-services = "sv status /service/*"
+services = "systemctl list-units --type=service"
 
-syslog = "tail -n 100 /var/log/messages"
+syslog = "journalctl -n 100"
 
-reboot = "/sbin/reboot 2>&1"
\ No newline at end of file
+reboot = "/usr/bin/reboot 2>&1"

--- webconfig/guinnux_web_base/srv/http/config/time_conf.lua    2015-12-23 16:23:24.052720647 +0200
+++ webconfig/guinnux_web_base/srv/http/config/time_conf-new.lua        2015-12-23 16:26:31.880193546 +0200
@@ -16,11 +16,11 @@
 
 hwclock_show = "/sbin/hwclock --show -u"
 
-ntp_start = "/usr/bin/sv up /service/ntpd"
+ntp_start = "systemctl start  ntpd"
 
-ntp_stop = "/usr/bin/sv down /service/ntpd"
+ntp_stop = "systemctl stop ntpd"
 
-ntp_restart = "service ntpd restart"
+ntp_restart = "systemctl restart ntpd"
 
 get_year = "date +%Y"
                  
--- webconfig/guinnux_web_base/srv/http/network_status.lua	2015-12-24 11:51:24.120500793 +0200
+++ webconfig/guinnux_web_base/srv/http/network_status-new.lua	2015-12-28 10:38:26.168471270 +0200
@@ -27,10 +27,10 @@
                     <!-- routing end -->
                     </pre>
 
-                    <h3>Interfaces file: </h3>
+                    <h3>Network Profiles: </h3>
                     <pre class="formLayoutfull">
                     <!-- interfaces start -->
-]]..command.ex("cat "..network_conf.interface_file )..[[
+]]..command.ex("netctl list")..[[
                     <!-- interfaces end -->
                     </pre>
 
